packages/@lwc/integration-karma/test/rendering/callback-invocation-order/index.spec.js

```diff
@@ -228,3 +229,54 @@ it('should invoke callbacks on the right order (issue #1199 and #1198)', () => {
         );
     });
 });
+
+it('should invoke callbacks on the right order when multiple templates are used with lwc:if', () => {
+    const elm = createElement('x-multi-template-conditionals', { is: MultiTemplateConditionals });
+    elm.show = true;
+    document.body.appendChild(elm);
+
+    // initial load is x-shadow-parent
+    expect(window.timingBuffer).toEqual([
+        'leaf:T1-1:connectedCallback',
+        'leaf:T1-2:connectedCallback',
+        'leaf:T1-3:connectedCallback',
+        'leaf:T1-4:connectedCallback',
+        'leaf:T1-5:connectedCallback',
+        'leaf:T1-6:connectedCallback',
+    ]);
+
+    resetTimingBuffer();
+    elm.next();
+
+    return Promise.resolve()
+        .then(() => {
+            // disconnect x-shadow-parent +
+            // connect x-shadow-container with 2 parents, 'a' and 'b'
+            expect(window.timingBuffer).toEqual([
+                'leaf:T1-6:disconnectedCallback',
+                'leaf:T1-5:disconnectedCallback',
+                'leaf:T1-4:disconnectedCallback',
+                'leaf:T1-3:disconnectedCallback',
+                'leaf:T1-2:disconnectedCallback',
+                'leaf:T1-1:disconnectedCallback',
+                'leaf:T2-1:connectedCallback',
+                'leaf:T2-2:connectedCallback',
+                'leaf:T2-3:connectedCallback',
+                'leaf:T2-4:connectedCallback',
+                'leaf:T2-5:connectedCallback',
+                'leaf:T2-6:connectedCallback',
+            ]);
+            resetTimingBuffer();
+            elm.show = false;
+        })
+        .then(() => {
+            expect(window.timingBuffer).toEqual([
+                'leaf:T2-6:disconnectedCallback',
```

Comment:
This step, hiding the content, is interesting because it doesn't follow the same code path as when the template is swapped.

Again, I checked the ordering in both native and synthetic:
```
if:true - synthetic - 654321
lwc:if - synthetic  - 123456
if:true - native - 654321
lwc:if - native - 123456
```
Here, `if:true` is always managed by `updateStaticChildren`, which goes in reverse order.
`lwc:if` is always managed by `unmountVNodes`, which goes in forward order.

I believe it's too late and risky to try to change the behavior for `if:true`, so we have to decide whether we want `lwc:if` to have the correct forward behavior in both synthetic and native, or figure out how to change it in synthetic to be consistent.

Reply:
Note that changing the ordering for lwc:if here would be difficult since it goes through `unmountVNodes`.

Reply:
Conclusion from dev sync: leave as-is, as we don't guarantee any expectations around ordering of disconnectedCallback when it comes to sibling nodes.